nvaccess/nvda
# Make font size measurements translatable

### Link to issue number:

Fixes #13573

### Summary of the issue:
"pt" when reporting font size measurements is not translatable.
Font size measurements are reported inconsistently, for example in excel and PowerPoint are reported as "11.0" instead of "11 pt".

### Description of how this pull request fixes the issue:
Makes the font size measurement "pt" translatable across APIs.

Browsers provide "font-size" attribute directly as "12pt|px|em|%|rem|ex"  or "large", "small", etc.
This requires a more strategic approach for catching each font-size case and replacing it with a translated string.
Documentation has been added for this.
https://developer.mozilla.org/en-US/docs/Web/CSS/font-size

### Testing strategy:
Unit testing has been added to test the more complicated parsing of the browser font-size attribute.

Manual testing

- Enable "font size" reporting in "Document Formatting".
- Added a translation to `/de/nvda.po`.
    ```po
    msgctxt "font size"
    msgid "%s pt"
    msgstr "%s Punkt"
    ```
- Switched NVDA language to German.
- Confirm that font-size is reported as "xx Punkt" when navigating text in:
  - [ ] adobe reader (unable to test due to x64 bug, 774f1ea has been tested)
  - [x] word (with and without UIA)
  - [x] excel (with and without UIA)
  - [x] powerpoint
  - [x] Wordpad
  - [x] Libre Office Writer (soffice)
  - [x] Notepad++ (scintilla)
  - [ ] JAB (unable to test, unknown relevant software)
  - [x] MSHTML (tested with IE mode for Edge on Win 11)
  - [x] IAccessible (Chrome, Firefox) 

### Known issues with pull request:
None

### Change log entries:
Changes
```- Font size measurements are now translatable in NVDA. (#13573)```

### Code Review Checklist:



  - description is up to date
  - change log entries
  - Unit tests
  - System (end to end) tests
  - Manual testing
  - User Documentation
  - Developer / Technical Documentation
  - Context sensitive help for GUI changes
  - Speech 
  - Braille
  - Low Vision
  - Different web browsers
  - Localization in other languages / culture than English
